app/src/main/java/edu/agh/susgame/front/ui/component/menu/CreateGameView.kt

```diff
@@ -1,32 +1,182 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
 import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Slider
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+
 import androidx.navigation.NavController
 import edu.agh.susgame.front.ui.Translation
 import edu.agh.susgame.front.ui.component.common.Header
 import edu.agh.susgame.front.ui.component.menu.navigation.MenuRoute
 import edu.agh.susgame.front.ui.theme.PaddingL
 
+import java.util.Locale
+import kotlin.math.roundToInt
+
+
+private const val maxPinLength = 6
+private const val maxPlayersAmount = 6
+private const val minPlayersAmount = 2
+private const val defaultPlayersAmount = 4
+private const val minGameTime = 5f
+private const val maxGameTime = 15f
+private const val defaultGameTime = 10
 @Composable
 fun CreateGameView(
     navController: NavController,
 ) {
-    Column(Modifier.padding(PaddingL)) {
+    var gameName by remember { mutableStateOf(Translation.CreateGame.DEFAULT_GAME_NAME) }
+    var gamePIN by remember { mutableStateOf("") }
+    var showPassword by remember { mutableStateOf(value = false) }
+    var selectedNumber by remember { mutableIntStateOf(defaultPlayersAmount) }
+    var gameTime by remember { mutableIntStateOf(defaultGameTime) }
+    Column( Modifier.padding(PaddingL)) {
         Column(
             Modifier
-                .fillMaxHeight()
+                .padding(PaddingL)
                 .weight(1f)
         ) {
             Header(title = Translation.Menu.CREATE_GAME)
 
-            // TODO GAME-52
-            Text(text = "TODO GAME-52")
+            // entering game name
+            OutlinedTextField(
+                label = { Text(Translation.CreateGame.ENTER_GAME_NAME) },
+                value = gameName,
+                onValueChange = { gameName = it },
+                modifier = Modifier.fillMaxWidth(),
+                singleLine = true
+            )
+
+            // entering game pin
+            OutlinedTextField(
```

Comment:
Myślałem, żeby zrobić `Switch` z podpisem typu "Ustaw PIN do dołączenia" i tylko gdy to jest wybrane, to pojawia się input na PIN. Nie jest to w sumie na razie istotne, więc możesz na razie zostawić jak jest

Reply:
bardzo spoko pomysł, dodałem narazie mały checkbox

Reply:
Spoko spoko